### DISCUSSION TITLE
cleanup not called when ancestor element removed

### DISCUSSION THREAD

**Discussion by ayamomiji** [alpinejs/alpine#1567] [7. Bugs (archived)] [answered]
version: 3.0.8

examples:
https://codepen.io/ayamomiji/pen/gOmyaEL?editors=1111

**Comment by calebporzio** [comment, marked as answer]
Good catch, there needs to be a more robust system that walks a removed element's children and cleans them up as well.

**Reply by calebporzio** [reply]
Currently, there are WeakMaps of elements and their cleanup functions - instead I think a better system would be attaching the cleanup functions directly to the dom nodes and running them as we walk

**Reply by calebporzio** [reply]
Fixed in this PR: https://github.com/alpinejs/alpine/pull/1569

Will be in the next release 👌
